stop-mycroft.sh

```diff
@@ -40,37 +41,37 @@ function help() {
     exit 0
 }
 
-function process-running() {
-    if [[ $( pgrep -f "python3 (.*)-m mycroft.*${1}" ) ]] ; then
+process_running() {
+    if [ "$( pgrep -f "python3 (.*)-m mycroft.*${1}" )" ] ; then
         return 0
     else
         return 1
     fi
 }
 
-function end-process() {
-    if process-running $1 ; then
+end_process() {
+    if process_running "$1" ; then
         # Find the process by name, only returning the oldest if it has children
         pid=$( pgrep -o -f "python3 (.*)-m mycroft.*${1}" )
-        echo -n "Stopping $1 (${pid})..."
-        kill -SIGINT ${pid}
+        printf "Stopping %s (%s)..." "$1" "${pid}"
+        kill -SIGINT "${pid}"
 
         # Wait up to 5 seconds (50 * 0.1) for process to stop
         c=1
         while [ $c -le 50 ] ; do
-            if process-running $1 ; then
+            if process_running "$1" ; then
                 sleep 0.1
-                (( c++ ))
+                c=$((c + 1))
             else
                 c=999   # end loop
             fi
         done
 
-        if process-running $1 ; then
+        if process_running "$1" ; then
             echo "failed to stop."
             pid=$( pgrep -o -f "python3 (.*)-m mycroft.*${1}" )            
-            echo -n "  Killing $1 (${pid})..."
-            kill -9 ${pid}
+            printf "  Killing %s (%s)..." "$1" "${pid}"
```

Comment:
```
printf "  Killing %s (%s)..." "${1}" "${pid}"
```

also, when I tried this I don't know why it trails a `$`

```
$ FOO=blah
$ BAR=23

$ printf "  Killing %s (%s)..." "$FOO" "$BAR"
  Killing blah (23)...$ ```

Reply:
Is that your prompt maybe?

There should be a `\n` at the end of the string

Reply:
Strange, Bash doesn't seem to append the `\n` but zsh does. I'll add a `\n` anyway to force it.